rcl/test/rcl/test_graph.cpp

```diff
@@ -946,25 +946,23 @@ class NodeGraphMultiNodeFixture : public CLASSNAME(TestGraphFixture, RMW_IMPLEME
 
   void WaitForAllNodesAlive()
   {
-    rcl_ret_t ret;
-    rcutils_string_array_t node_names = rcutils_get_zero_initialized_string_array();
-    rcutils_string_array_t node_namespaces = rcutils_get_zero_initialized_string_array();
-    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
-    {
-      ret = rcutils_string_array_fini(&node_names);
-      ASSERT_EQ(RCUTILS_RET_OK, ret);
-      ret = rcutils_string_array_fini(&node_namespaces);
-      ASSERT_EQ(RCUTILS_RET_OK, ret);
-    });
     // wait for all 3 nodes to be discovered: remote_node, old_node, node
     size_t attempts = 0;
-    size_t max_attempts = 4;
-    while (node_names.size < 3) {
+    size_t max_attempts = 10;
+    rcutils_string_array_t node_names = rcutils_get_zero_initialized_string_array();
+    rcutils_string_array_t node_namespaces = rcutils_get_zero_initialized_string_array();
+    do {
       std::this_thread::sleep_for(std::chrono::seconds(1));
-      ret = rcl_get_node_names(this->remote_node_ptr, allocator, &node_names, &node_namespaces);
+      node_names = rcutils_get_zero_initialized_string_array();
+      node_namespaces = rcutils_get_zero_initialized_string_array();
+      ASSERT_EQ(
+        RCL_RET_OK,
+        rcl_get_node_names(this->remote_node_ptr, allocator, &node_names, &node_namespaces));
       attempts++;
+      ASSERT_EQ(RCUTILS_RET_OK, rcutils_string_array_fini(&node_names));
+      ASSERT_EQ(RCUTILS_RET_OK, rcutils_string_array_fini(&node_namespaces));
```

Comment:
Why not leave this in a scope exit?

Reply:
I don't see a big benefit of using that here.

Before we were doing so, and the test was still leaking because of passing the same string array `node_names` and `node_namespaces` multiple times to `rcl_get_node_names()`.

Now we're calling `rcutils_string_array_fini()` immediately after the function that makes the allocation, so the scope exit macro does not seem to add much value.

Reply:
Oh I guess there's no reason to not get to these lines if `rcl_get_node_names` succeeds, I was thinking like if someone added an assertion between lines 959 and 962, it would leak if that assertion failed.

But as-is it is fine.